eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/http/RequestUriAndTimeLogging.java

```diff
@@ -56,8 +59,16 @@ public RequestUriLogging(HttpRequestExecutor delegate, boolean enableLog, String
     {
         if (mEnableLog)
         {
-            Log.d(mTag, String.format("%s %s", request.method().verb(), uri));
+            int requestId = sRequestCounter.getAndIncrement();
+            Log.d(mTag, String.format("(%s->) %s %s", requestId, request.method().verb(), uri));
+            long start = SystemClock.uptimeMillis();
+            T result = mDelegate.execute(uri, request);
+            Log.d(mTag, String.format("(%s<-) Response time: %s ms", requestId, SystemClock.uptimeMillis() - start));
+            return result;
+        }
+        else
+        {
+            return mDelegate.execute(uri, request);
```

Comment:
With the 2 branches it ends up like this:
```
        int requestId = 0;
        long start = 0;
        if (mEnableLog)
        {
            requestId = sRequestCounter.getAndIncrement();
            Log.d(mTag, String.format("(%s->) %s %s", requestId, request.method().verb(), uri));
            start = SystemClock.uptimeMillis();
        }

        T result = mDelegate.execute(uri, request);

        if (mEnableLog)
        {

            Log.d(mTag, String.format("(%s<-) Response time: %s ms", requestId, SystemClock.uptimeMillis() - start));
        }
        
        return result;
```
Not very nice either, I think. So now I did the routing in the constructor for the delegation, check it out.